### PR TITLE
Ci improvements docker images

### PR DESCRIPTION
**Pull request by marcinwrobel1986**
- added dockerfiles
- built docker images
- sent docker images to the dockerhub repository
- started CI cancelling action
- added testing for current script with usage of created docker images

### CODE REVIEW

File: .github/workflows/pressio_testing.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      gtest_path: /usr/local/gtest/install
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build pressio tests
+        run: |
+          cd /pressio-builder
+          git pull
+          git status
+          sed -i 's/source .\/shared\/colors.sh/# colors commnted out/g' main_pressio.sh
+          ./main_pressio.sh -dryrun=no -build-mode=Release -target-dir=${{env.target_dir}} -eigen-path=${{env.eigen_path}} -gtest-path=${{env.gtest_path}} -cmake-generator-name=default_with_tests -pressio-src=${{env.pressio_dir}}
```
Comment (fnrizzi) [inline .github/workflows/pressio_testing.yml:35]: @marcinwrobel1986 before merging this, can you please add also builds for "debug" ? for all images, I would like both Release and Debug - thanks

**Review by fnrizzi — COMMENTED** [review]
I left one comment about adding debug builds too please

File: .github/workflows/pressio_testing.yml (hunk 1 above)
Comment (marcinwrobel1986) [reply .github/workflows/pressio_testing.yml:35]: Ok, will do. I will let you know, when done.

Comment (fnrizzi) [reply .github/workflows/pressio_testing.yml:35]: thanks! no rush!

**Pull request merged**